### ISSUE TITLE
Add schema for [tool.pdm] table to schemastore.org

### ISSUE THREAD

**Issue by Elijahh177** [pdm-project/pdm#2490]
## Is your feature request related to a problem? Please describe.

When adding properties into the `[tool.pdm]` table, no intellisense is available (VS Code + Even Better TOML extension). However, intellisense is available for `[tool.poetry]`.

## Describe the solution you'd like

The VS Code extension [Even Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml) validates the `pyproject.toml` with the help of a JSON schema from [schemastore.org](https://www.schemastore.org). A `pdm.json` schema could be added to schemastore to enable validation of the `[tool.pdm]` table.

@frostming Do you already have a schema for PDM? Otherwise, I would be happy to contribute one🙂

**Comment by frostming** [comment]
> @frostming Do you already have a schema for PDM? Otherwise, I would be happy to contribute one🙂

Nope, appreciate for your help

**Comment by jvllmr** [comment]
I also gave this a try and I have more or less finished with the `pdm.toml` file.

Next topic is the actual `tool.pdm` table in the `pyporject` file which has much more meta-rules.

I already created an example file with most of the config keys and values I could find in the documentations of pdm and pdm-backend: https://github.com/jvllmr/schemastore/blob/master/src/test/pyproject/pdm_tool.toml

Let me know if something is missing or incorrect.
